pkg/cmd/pipelinerun/delete.go

```diff
@@ -80,13 +82,16 @@ func deletePipelineRuns(s *cli.Stream, p cli.Params, prNames []string, opts *opt
 	if err != nil {
 		return fmt.Errorf("failed to create tekton client")
 	}
+	if opts.Keep > 0 && !opts.DeleteAllNs {
```

Comment:
This error message will only be reached if using `-f`. My recommendation would be to check earlier in `RunE` or `Args` to account for prompt situation:

If I run `tkn pr delete --keep 7`, I get the following:

```
Error: must provide pipelinerun name(s) or use --pipeline flag or --all flag to use delete
```